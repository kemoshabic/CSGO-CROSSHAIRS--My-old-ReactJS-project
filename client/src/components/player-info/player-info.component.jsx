import React, { useState } from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import Settings from "../settings-overview/settings-overview.component";
import InfoButton from "../info-buton/info-button.component";
import Copy from "../Copy/copy.component";

import { selectPlayersInfo } from "../../redux/teams/teams.selector";
import { togglePlayersInfoHidden } from "../../redux/teams/teams.actions";
import {
  addCrosshair,
  toggleSavedButton,
  addSettings,
} from "../../redux/settings/settings.actions";
import {
  selectIfCopied,
  selectIfSaved,
} from "../../redux/settings/settings.selector";

import "./player-info.styles.scss";

import Target from "../../assets/target.png";
import Computer from "../../assets/computer.png";
import RedHeart from "../../assets/pink.png";
import PurpleHeart from "../../assets/purple.png";
import Close from "../../assets/close.png";
import WhiteClose from "../../assets/WhiteClose.png";

import  Img  from 'react-graceful-image'

export const PlayerInfo = ({
  info: {
    name,
    img,
    settings,
    role,
    crosshair,
    team,
    id,
    crosshairImg,
    imageUrl,
  },
  togglePlayersInfoHidden,
  copied,
  addCrosshair,
  toggleSavedButton,
  saved,
  addSettings,
}) => {
  const savingData = () => {
    toggleSavedButton(true) && setTimeout(() => toggleSavedButton(false), 1200);
  };
  const [shown, setShown] = useState(false);
  const [shownCrosshair, setShownCrosshair] = useState(false);
  return (
    <div className="info">
      {shownCrosshair ? (
        <div className="info__crosshair-overview">
          <div className="info__crosshair-preview">
            <img
              className="info__crosshair-btn"
              src={WhiteClose}
              alt="close button"
              onClick={() => setShownCrosshair(!shownCrosshair)}
            />
            <img
              className="info__crosshair-img"
              src={crosshairImg}
              alt="crosshair Img"
            />
          </div>
        </div>
      ) : null}
      <div
        className={`${
          copied || saved ? "info__copied-succes" : ""
        } info__copied`}
      >
        <p className="info__copied-text">
          {saved ? "Saved" : "Copied to clipboard"}
        </p>
      </div>

      <div className="info__header">
        <h3 className="info__heading">{name}</h3>
        <img
          src={Close}
          alt="btn"
          className="info__close"
          onClick={togglePlayersInfoHidden}
        />
      </div>
      <div className="info__grid">
        <div className="info__section">
          <Img src={img} alt="img" className="info__player" />
          <div className="info__overview">
            <span className="info__role">{role}</span>
            <div className="info__wrap">
              <div
                className="info__crosshair-wrap"
                onClick={() => setShownCrosshair(!shownCrosshair)}
              >
                <InfoButton img={Target} text={"Crosshair preview"} />
              </div>
              <div
                className="info__save"
                onClick={() =>
                  addCrosshair({ team, name, crosshair, id, imageUrl }) &&
                  savingData()
                }
              >
                <img src={RedHeart} alt="copy" />
                <p className="info__text">Save Crosshair settings</p>
              </div>
              <InfoButton img={Computer} anchor text={"Players Resolution"} />
              <div
                className="info__save"
                onClick={() =>
                  addSettings({ team, name, id, settings }) && savingData()
                }
              >
                <img src={PurpleHeart} alt="copys" />
                <p className="info__text">Save User settings</p>
              </div>
            </div>
          </div>
        </div>

        <Settings settings={settings} />
        <div className="info__footer">
          <Copy crosshair={crosshair} />
          <p className="info__config" onClick={() => setShown(!shown)}>
            See crosshair config
          </p>
          <div className={`${shown ? "info__show--toggle" : ""} info__show`}>
            <p className="info__show--text">{crosshair}</p>
          </div>
        </div>
      </div>
    </div>
  );
};
const mapStateToProps = createStructuredSelector({
  info: selectPlayersInfo,
  copied: selectIfCopied,
  saved: selectIfSaved,
});

const mapDispatchToProps = (dispatch) => ({
  togglePlayersInfoHidden: () => dispatch(togglePlayersInfoHidden()),
  addCrosshair: (item) => dispatch(addCrosshair(item)),
  toggleSavedButton: (boolVal) => dispatch(toggleSavedButton(boolVal)),
  addSettings: (item) => dispatch(addSettings(item)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PlayerInfo);
