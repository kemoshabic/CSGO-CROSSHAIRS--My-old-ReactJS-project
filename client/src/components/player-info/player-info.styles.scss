@mixin positionAbsoluteAndCenter{
  position: absolute;
  top: 50%;
  left: 50%;                    
  transform: translate(-50%, -50%);
}
@mixin displayFlexAndDirection{
  display: flex;
  flex-direction: column;
}
.info{
  @include positionAbsoluteAndCenter();
  @include displayFlexAndDirection();
  z-index: 25;
  width: 130rem;
  height: 83rem;
  background-color: #ffffff;
  position: fixed;

  overflow-y: auto;
  overflow-x: hidden;
  @media only screen and (max-width: 68.75em) {
    // 1100px
    width: 115rem;
    height: 70rem;
  }

  @media only screen and (max-width: 57.5em) {
    width: 100rem;
    height: 80rem;
  }

  @media only screen and (max-width: 50em) {
    // 800px
    width: 62rem;
    height: auto;
    overflow-y: scroll;
    max-height: 100%;
  }

  @media only screen and (max-width: 28.12em) {
    width: 50rem;
  }

  
  &__header{
    display: flex;
    padding: 2.2rem 3.2rem;
    align-items: center;
    border-bottom: 1px solid #0A2B47;
    @media only screen and (max-width: 68.75em) {
      // 1100px
      padding: 1rem 3.2rem;
    }
  }
  &__heading{
    color: #0A2B47;
    font-size: 2.8rem;
    font-weight: 400;
    margin-right: auto;
  }
  &__close{
    width: 2.4rem;
    height: 2.4rem;
    cursor: pointer;
  }

  &__grid{
    padding: 6rem 11rem;
    display: grid;
    align-self: start;
    @media only screen and (max-width: 68.75em) {
      // 1100px
      padding: 3rem 5.5rem;
    }
  }
  &__section{
    display: flex;
    margin-bottom: 4rem;
    @media only screen and (max-width: 50em) {
      // 800px
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
  }
  &__player{
    width: 35rem;
    height: 25rem;
    margin-right: 6rem;
    object-fit: cover;
  }
  &__overview{
    @include displayFlexAndDirection();
    padding-top: 4rem;
  }
  &__role{
    background-color: #FD074E;
    font-size: 1.6rem;
    border-radius: 2.5rem;
    width: 6rem;
    height: 3.2rem;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 2rem;
  }

  &__wrap{
    display: grid;
    grid-template-rows: 4.5rem 4.5rem;
    grid-template-columns: 27rem 27rem;
    grid-row-gap: 1.7rem;
    grid-column-gap: 2rem;

    @media only screen and (max-width: 28.12em) {
      grid-row-gap: 1rem;
      grid-column-gap: 0.6rem;
      grid-template-rows: repeat(4, 4.5rem);
      grid-template-columns: 27rem;
      align-items: center;
      justify-content: center;
    }

    
  }
  &__crosshair-overview{
    position: absolute;
    width:130rem ;
    height: 83rem;
    background-color: rgba(#000000, 0.8);
    overflow: hidden;
    z-index: 2;
    @media only screen and (max-width: 68.75em) {
      // 1100p
      width:120rem;
      background-color: transparent;
    }

    @media only screen and (max-width: 50em) {
      // 800px
      width: 65rem;
    }

    @media only screen and (max-width: 25em) {
      // 800px
      width: 50rem;
    }
  }

  &__crosshair-preview{
    width: 44rem;
    height: 33rem;
    background-color: #0A0B20;
    @include positionAbsoluteAndCenter();
    @include displayFlexAndDirection();
    align-items: flex-end;
    border-radius: 2.5rem;
  }
  
  &__crosshair-img{
    align-self: center;
    margin-top: 1rem;
    @media only screen and (max-width: 81.25em) {
      // 1300px
      width: 32rem;
      height: 25rem;
    }
    @media only screen and (max-width: 68.75em) {
      // 1100px
      width: 30rem;
      height: 20rem
    }
  }
  &__crosshair-btn{
    width: 2rem;
    height: 3rem;
    padding-top :1rem;
    margin-right: 1.5rem;
    cursor: pointer;
  }

  &__save{
    display: flex;
    align-items: center;
    cursor:pointer;
    @media only screen and (max-width: 28.12em) {
     justify-content: center;
    }
  }

  &__text{
    color: #000000;
    font-size: 1.8rem;
    padding-left: 1.5rem;
    &:hover{
      color: #FD074E;
    }
  }
  &__footer{
    padding-top:4rem;
    display: flex;
    align-items: flex-start;
    @media only screen and (max-width: 50em) {
      // 800px
      flex-direction: column;
    }
  }
  &__show{
    margin-left: 2rem;
    margin-top: -2rem;
    border-radius: 0.5rem;
    width: 35rem;
    height: 21em;
    background-color: #FD074E;
    display: none;
    box-shadow: 0 1.5rem 4rem rgba(#000000, 0.15);
    @media only screen and (max-width: 68.75em) {
      // 1100px
      width: 50rem;
      height: 16rem;
      margin-left: 1rem;
    }

    @media only screen and (max-width: 28.12em) {
      width: 40rem;
      height: 20rem;
    }

    &--text{
      padding: 1rem 2rem;
      font-size: 1.5rem;
    }
    &--toggle{
      display: block;
    }
  }
  &__config{
    margin-top: 1.7rem;
    padding-left: 2rem;
    color: #FD074E;
    font-size: 1.8rem;
    text-decoration: underline;
    cursor: pointer;
    @media only screen and (max-width: 50em) {
      padding-bottom: 5rem;
    }
  }
  &__copied{
    color: black;
    position: absolute;
    bottom: 2rem;
    right: 2rem;
    display: none;
    width: 25rem;
    height: 6rem;
    background-color: #21AC54;
    font-size: 2.2rem;
    display: none;
    align-items: center;
    justify-content: center;
    border-radius: 2.5rem;
    @media only screen and (max-width: 68.75em) {
      width: 20rem;
      height: 5rem;
      font-size: 1.8rem;
      bottom: 0;
    }
    &-text{
      color: #ffffff;
    }
    &-succes{
      display: flex;
    }
  }

}
