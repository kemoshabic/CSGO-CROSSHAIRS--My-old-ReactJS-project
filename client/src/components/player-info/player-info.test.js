import React from "react";
import { shallow } from "enzyme";

import { PlayerInfo } from "./player-info.component";

describe("PlayerInfo component", () => {
  let wrapper;
  let mockTogglePlayersInfoHidden;
  let mockAddCrosshair;

  const mockInfo = {
    id: 1,
    name: "text",
    img: "jpg",
    role: "text",
    crosshairImg: "jpg",
    settings: {
      monitor: "a",
      mouse: "b",
      keyboard: "c",
      headset: "d",
      mousepad: "e",
    },
    crosshair: "text",
  };

  beforeEach(() => {
    mockTogglePlayersInfoHidden = jest.fn();
    mockAddCrosshair = jest.fn();

    const mockProps = {
      info: mockInfo,
      togglePlayersInfoHidden: mockTogglePlayersInfoHidden,
      addCrosshair: mockAddCrosshair,
    };

    wrapper = shallow(<PlayerInfo {...mockProps} />);
  });

  it("should render PlayerInfo component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render not render CrosshairOverview by default", () => {
    expect(wrapper.exists(".info__crosshair-overview")).toBe(false);
  });

  it("should call addCrosshairwhen button is clicked", () => {
    wrapper.find(".info__wrap").childAt(1).simulate("click");
    expect(mockAddCrosshair).toHaveBeenCalled();
  });
  
  it("should call togglePlayersInfoHidden when button is clicked", () => {
    wrapper.find(".info__close").simulate("click");
    expect(mockTogglePlayersInfoHidden).toHaveBeenCalled();
  });
  it("should return data values properly", () => {
    const name = wrapper.find(".info__heading").text();
    expect(name).toBe("text");
    const role = wrapper.find(".info__role").text();
    expect(role).toBe("text");
    const crosshair = wrapper.find(".info__show--text").text();
    expect(crosshair).toBe("text");
  });
});
