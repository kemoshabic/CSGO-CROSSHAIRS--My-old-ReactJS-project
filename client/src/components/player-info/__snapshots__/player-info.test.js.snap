// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInfo component should render PlayerInfo component 1`] = `ShallowWrapper {}`;
